kinoshita/laravel-study/miniblog/app/Http/Controllers/DashboardController.php

```diff
@@ -26,7 +26,12 @@ public function index()
 
         Debugbar::info($all_posts);
 
-        return view('dashboard', compact('all_posts'));
+        $show_comment_btn = true;
+
+        return view('dashboard', [
+            "all_posts" => $all_posts,
+            "show_comment_btn" => $show_comment_btn
+        ]);
```

Comment:
無駄のコード多いので、以下のどっちかにしてほしいです

```suggestion
        $show_comment_btn = true;

        return view('dashboard', compact(["all_posts", "show_comment_btn"]) );
```

```suggestion
        return view('dashboard', [
            "all_posts" => $all_posts,
            "show_comment_btn" => true
        ]);
```

※Debugbarをなくす場合、

- `$all_posts=Post::with(['user', 'comments'])->get();`もなくして
- `"all_posts" => Post::with(['user', 'comments'])->get(),`にする

方が、読みやすいです。

Reply:
DebagBarは一旦残して、compactを使いました  

```
  public function index()
    {
        $all_posts = Post::with(['user', 'comments'])->get();

        Debugbar::info($all_posts);

        $show_comment_btn = true;

        return view('dashboard', compact(['all_posts', 'show_comment_btn']));
    }

```